### PR TITLE
PS-9533 [8.0]: Introduce cmake option to enable probuilds

### PR DESCRIPTION
**Pull request by inikep**
Add `WITH_PERCONA_PROBUILD` cmake option that enables pro features. 
The`PROBUILD` cmake option is an alias for `WITH_PERCONA_PROBUILD`. 
From source code we have to use `#ifdef HAVE_PERCONA_PROBUILD` for pro features.

### CODE REVIEW

**Comment by inikep** [comment]
```
./bin/mysqld-debug --version
/data/mysql-server/percona-8.0-deb-gcc14-add/runtime_output_directory/mysqld-debug  Ver 8.0.40-31-pro-debug for Linux on x86_64 (Source distribution)
```
```
SELECT @@version;
@@version
8.0.40-31-pro-debug
```
```
/data/mysql-server/percona-8.0-deb-gcc14-add/runtime_output_directory/mysqld-debug --no-defaults --initialize-insecure --loose-skip-ndbcluster --tmpdir=/data/mysql-server/percona-8.0-deb-gcc14-add/mysql-test/var/tmp/ --core-file --datadir=/data/mysql-server/percona-8.0-deb-gcc14-add/mysql-test/var/data/ --secure-file-priv=/data/mysql-server/percona-8.0-deb-gcc14-add/mysql-test/var --innodb_buffer_pool_size=24M --innodb-redo-log-capacity=10M --innodb_autoextend_increment=8 --character-sets-dir=/data/mysql-server/percona-8.0/share/charsets --loose-auto_generate_certs=OFF --loose-sha256_password_auto_generate_rsa_keys=OFF --loose-caching_sha2_password_auto_generate_rsa_keys=OFF --init-file=/data/mysql-server/percona-8.0-deb-gcc14-add/mysql-test/var/tmp/bootstrap.sql
2024-12-04T11:59:53.704362Z 0 [System] [MY-013169] [Server] /data/mysql-server/percona-8.0-deb-gcc14-add/runtime_output_directory/mysqld-debug (mysqld 8.0.40-31-pro-debug) initializing of server in progress as process 727439
2024-12-04T11:59:53.728987Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-12-04T11:59:55.761935Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
```

File: configure.cmake [hunk 1]
```diff
@@ -556,6 +556,10 @@ IF(WITH_PERCONA_TELEMETRY)
   SET(HAVE_PERCONA_TELEMETRY 1)
 ENDIF()
 
+IF(WITH_PERCONA_PROBUILD OR PRO_BUILD)
+  SET(HAVE_PERCONA_PROBUILD 1)
+ENDIF()
+
```
Comment (satya-bodapati) [inline configure.cmake]: this is waht I did for pxb
```
+OPTION(PROBUILD "Build Percona XtraBackup Pro variant" OFF)
+IF (PROBUILD)
+ SET(XTRABACKUP_SUFFIX "-pro")
+ ADD_DEFINITIONS(-DPROBUILD)
+ENDIF()
```

File: sql/mysqld.cc [hunk 2]
```diff
@@ -11838,8 +11838,13 @@ static int get_options(int *argc_ptr, char ***argv_ptr) {
 static void set_server_version(void) {
   char *end [[maybe_unused]] = strxmov(server_version, MYSQL_SERVER_VERSION,
                                        MYSQL_SERVER_SUFFIX_STR, NullS);
+#ifdef HAVE_PERCONA_PROBUILD
+  if (!strstr(MYSQL_SERVER_SUFFIX_STR, "-pro"))
+    end = my_stpcpy(end, "-pro");
+#endif
```
Comment (satya-bodapati) [inline sql/mysqld.cc]: This change is not required if we define this in CMake and the above code will take care of this

File: configure.cmake (hunk 1 above)
Comment (satya-bodapati) [reply configure.cmake]: Also I used the cmake option PROBUILD. Better to use same name across all products. 

Or if you feel -DPERCONA_PROBUILD=1 is better, I will change it in next release.

@percona-ysorokin your thoughts?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: configure.cmake (hunk 1 above)
Comment (inikep) [reply configure.cmake]: @satya-bodapati I followed `WITH_VALGRIND` and `WITH_PERCONA_TELEMETRY` style.
I also added `PROBUILD` option (in Jira there was `PRO_BUILD` but probably you changed it later) which is equal to `WITH_PERCONA_PROBUILD`.

Comment (satya-bodapati) [reply configure.cmake]: yeah, I changd it may be. so i am using -DPROBUILD=1 for cmake and within source #ifdef PROBUILD

**Review by percona-ysorokin — APPROVED** [review]
LGTM

File: configure.cmake [hunk 3]
```diff
@@ -556,6 +556,11 @@ IF(WITH_PERCONA_TELEMETRY)
   SET(HAVE_PERCONA_TELEMETRY 1)
 ENDIF()
 
+IF(WITH_PERCONA_PROBUILD OR PROBUILD)
+  SET(MYSQL_SERVER_SUFFIX "-pro")
+  SET(HAVE_PERCONA_PROBUILD 1)
```
Comment (satya-bodapati) [inline configure.cmake:561]: what is the difference between SET and ADD_DEFINITIONS?

Comment (inikep) [reply configure.cmake:561]: Probably you will not see `-DHAVE_PERCONA_PROBUILD` from command line because it will be defined in `config.h`.

Comment (satya-bodapati) [reply configure.cmake:561]: Ah, I see. I used ADD_DEFINITIONS and I see -DPROBUILD in command line.

**Pull request merged**